### PR TITLE
Add secret file credential type

### LINKED ISSUE #175: Adding secret file credential type
I need to be able to be able to access a secret and store the whole lot as json in a file. I am happy to work on this myself (though it might be slow as I haven’t worked on jenkins plugins before) but I’d appreciate any guidance on the best way to develop this.

Looking at the code, I think I can see how to add another function to VaultHelper. I was thinking rename `getVaultSecret` to something like `getVaultSecretKey` to keep the current uses, then have a  new `getVaultSecret` that grabs the whole secret and hands back the key:value map. Plus change the `SecretRetrieve` to get the whole secret and put the key filtering into the new `getVaultSecretKey` function. 
Does this sound like a good design?

### PR DESCRIPTION
**Pull request by jamesrobson-secondmind**
This adds the ability to create secret file credentials (equivalent to [org.jenkinsci.plugins.plaincredentials.FileCredentials](https://github.com/jenkinsci/plain-credentials-plugin/blob/master/src/main/java/org/jenkinsci/plugins/plaincredentials/FileCredentials.java)). The contents of the file can either be taken from a single key within a secret, or contain the whole secret as json.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

Fixes #175

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -201,6 +201,11 @@
       <classifier>tests</classifier>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>com.googlecode.json-simple</groupId>
+      <artifactId>json-simple</artifactId>
+      <version>1.1.1</version>
+    </dependency>
```
Comment (jetersen) [inline pom.xml:208]: I don't like this direct dependencies.

JSONObject from jenkins already exist: https://github.com/jenkinsci/json-lib/blob/16052d1f74908fb59430eddd73f42aec2469b313/src/main/jdk15/net/sf/json/JSONObject.java#L125 with a toString method

The com.bettercloud.vault-java-driver already comes with a JSON parser.

You could also depend on https://github.com/jenkinsci/jackson2-api-plugin

**Review by jetersen — CHANGES_REQUESTED** [review]

File: src/main/java/com/datapipe/jenkins/vault/credentials/common/VaultFileCredentialImpl.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+    @NonNull
+    @Override
+    public InputStream getContent() {
+        String content;
+        if (useKey != null && useKey) {
+            content = getVaultSecretKeyValue(vaultKey);
+        } else {
+            Map<String, String> s = getVaultSecretValue();
+            content = new JSONObject().fromObject(s).toString();
```
Comment (jetersen) [inline src/main/java/com/datapipe/jenkins/vault/credentials/common/VaultFileCredentialImpl.java:76]: A static method exist, no need to call new :) https://github.com/jenkinsci/json-lib/blob/16052d1f74908fb59430eddd73f42aec2469b313/src/main/jdk15/net/sf/json/JSONObject.java#L138-L140
```suggestion
            content = JSONObject.fromObject(s).toString();
```

**Comment by jamesrobson-secondmind** [comment]
The requested changes have been made and the tests are now passing, so can this be approved?

**Review by jetersen — APPROVED** [review]

**Comment by jetersen** [comment]
@jamesrobson-secondmind thanks for your contribution :)

**Pull request merged**